examples/beginner/example_nst_without_pystiche.py

```diff
@@ -444,30 +445,31 @@ def show_image(image, title=None):
 # . This structure is also valid for all other optimizers.
 
 num_steps = 500
+iterable = range(1, num_steps + 1)
 
-for step in range(1, num_steps + 1):
+with tqdm(desc="Image optimization", iterable=iterable) as progress_bar:
+    for _ in iterable:
 
-    def closure():
-        optimizer.zero_grad()
+        def closure():
+            optimizer.zero_grad()
 
-        input_encs = multi_layer_encoder(input_image, content_layers, style_layers)
-        input_content_encs, input_style_encs = input_encs
+            input_encs = multi_layer_encoder(input_image, content_layers, style_layers)
+            input_content_encs, input_style_encs = input_encs
 
-        content_score = content_loss(input_content_encs)
-        style_score = style_loss(input_style_encs)
+            content_score = content_loss(input_content_encs)
+            style_score = style_loss(input_style_encs)
 
-        perceptual_loss = content_score + style_score
-        perceptual_loss.backward()
+            perceptual_loss = content_score + style_score
+            perceptual_loss.backward()
 
-        if step % 50 == 0:
-            print(f"Step {step}")
-            print(f"Content loss: {content_score.item():.3e}")
-            print(f"Style loss:   {style_score.item():.3e}")
-            print("-----------------------")
+            progress_bar.set_postfix(
+                loss=f"{float(perceptual_loss):.3e}", refresh=False
+            )
```

Comment:
Just to confirm: we are not refreshing here, because we only update afterwards, right?

Reply:
Yes. If I'm not wrong, manual update was necessary since tqdm iterable was not a part of the loop.